### ISSUE TITLE
Bug: unknown error: Expecting value: line 1 column 1 (char 0)

### ISSUE THREAD

**Issue by DGrave** [JurajNyiri/HomeAssistant-Tapo-Control#29]
Describe the bug
unknown error trying to add the camera

To Reproduce
Installed from the repository the custom component, added the TAPO camera.

Expected behavior
unknown error trying to add the camera

Log
Log Details (ERROR)
Logger: custom_components.tapo_control.config_flow
Source: custom_components/tapo_control/config_flow.py:48
Integration: tapo_control (documentation, issues)
First occurred: 12:24:25 AM (1 occurrences)
Last logged: 12:24:25 AM
Expecting value: line 1 column 1 (char 0)

2020-10-15 00:24:25 ERROR (MainThread) [custom_components.tapo_control.config_flow] Expecting value: line 1 column 1 (char 0)

Camera (please complete the following information):
Device info
C200
by TP-Link
Firmware: 1.0.14 Build 200720 Rel.38552n(4555)

Additional context

**Comment by JurajNyiri** [comment]
Please look into files .storage/config_entries, core.device_registry, core.entity_registry and paste here anything that has tapo_control in its name.

Try to delete data for the camera, remove anything with tapo_control from .storage/config_entries, core.device_registry, core.entity_registry and restart.

How did you add the camera? Did all go well when adding it? When did error happen the first time?

**Comment by JurajNyiri** [comment]
I think this is happening when you are trying to add the camera. Is that correct?

What values are you putting in into ip address, username and password?

**Comment by DGrave** [comment]
Hi JurajNyiri

Yes that correct this is happening when I’m e trying to add the camera.
On mine my first tray yes, I add some special characters, but that was the first thing that I have changed, now I just have uppercase, thinks that’s ok.

I don’t have any thing related to tapo_control  on .storage/config_entries, core.device_registry, core.entity_registry.

I reinstalled tapo_control  several times, using HACS and manually bay Copy the components to custom_components/tapo_control/ .

I’m ruing Home assistant core in a docker inside UNRAID.

Thanks you

**Comment by JurajNyiri** [comment]
Hello, please send what you are putting into ip, username and password type

**Comment by DGrave** [comment]
Hi

IP: 20.20.20.73
User: CCTVFO
Passwrod type: Passwo44

**Comment by JurajNyiri** [comment]
Could you try again with the latest version?

If it doesn't work, try changing the user and password in tapo advanced settings to something really really simple, like "admin" and "abcdef". IP is 100% correct? Looks unusual.

I cannot replicate this. Anyone else experiencing this issue?

**Comment by JurajNyiri** [comment]
Closed because of inactivity.

**Comment by thundergreen** [comment]
I face same issue. I have the feling that the IP ist the problem here. oved my cams to a VLAN with IP like 40.40.40.xx and re-adding this cam would end up in same issue as described here.

**Comment by thundergreen** [comment]
> I think this is happening when you are trying to add the camera. Is that correct?
> 
> What values are you putting in into ip address, username and password (= does it have some special characters or something?)?

```
2020-11-13 12:51:12 ERROR (MainThread) [custom_components.tapo_control] Expecting value: line 1 column 1 (char 0)
```

![image](https://user-images.githubusercontent.com/13320319/99069630-05199800-25af-11eb-9baa-f3c6fd5a27a0.png)


![image](https://user-images.githubusercontent.com/13320319/99069664-16fb3b00-25af-11eb-9cf2-fa1525ee1fe0.png)


Camera rachable from Hass Host

**Comment by thundergreen** [comment]
Adding the camera manually like:

```
  - platform: ffmpeg
    input: rtsp://hassio:hassio@40.40.40.91:554/stream1
    name: Living Room
```
works pretty straight . So it's no connection issue but config lfow maybe as it expects 3 numbers fo the IP maybe?

![image](https://user-images.githubusercontent.com/13320319/99073051-78260d00-25b5-11eb-81c2-6e2b3d49062c.png)

**Comment by JurajNyiri** [comment]
I was able to replicate this by using IP 8.8.8.8. Working on fix.

**Comment by JurajNyiri** [comment]
While I was not able to find the cause (because of not having the vlan and testing with 8.8.8.8 - not tapo camera), I added better error log for this and many more scenarios. 
Please try again on 2.2.3 and let me know what shows up in the log.

**Comment by JurajNyiri** [comment]
We were debugging together with @thundergreen .

After updating to 2.2.3 the error we got was 403 - Forbidden. The same error was received when doing the request to the camera manually via curl.
What we have found out is that this is most probably a vlan issue.

Ports 443, 554 and 2020 need to be opened on affected vlans and available from/to Home Assistant to/from the Camera.

Enabling these ports depends on network setup and is out of scope of this integration.

I advised doing research on how to do that, or switching camera to the same vlan as Home Assistant is at, and (optionally) blocking its internet access there as that was working flawlessly.

I asked @thundergreen to update this issue when he discovers a solution that works for him.

Closing this issue as it is not related to this integration and a better error message is now shown in 2.2.3.
